### DISCUSSION TITLE
Error raised when using multi-DVGeo addPointSet()

### DISCUSSION THREAD

**Discussion by hntter** [mdolab/pygeo#257] [General]
Hi, Dear Developers,
I encountered some issues while using FFD for aerodynamic shape optimization of a bump on a cylindrical body. Initially, I used a single FFD block (named blcok1) to enclose the bump. However, when moving the control points, the connection surface between the bump and the cylinder became shows discontinuities or even negative volumes like pictures below.
![image](https://github.com/user-attachments/assets/a4c98b61-79c2-4ac0-bc2b-6e05a7f43dea)

Later, I reviewed some related papers ([Hannah Hajdik](https://www.researchgate.net/profile/Hannah_Hajdik?_sg%5B0%5D=oslQddQP10A90VQ8a_inukeD2fYiEoFwWie-ijo0IgZ1vDtOTjE0LW0t1BVMozENq5nJEsU.0DSUXH-3VViq0F-seAWeQaDvpgQEWnUv1CND1-K77j5I2e_YXbl89zKAUzWlQlcKr_dCRHEwgJs0KvMicwljpQ&_sg%5B1%5D=ZxPSARLXkBsEJH8BQ0ARTkF8uWGmGcg48s6rH7BtDaoDGzekRz--iLG9HpxsqnDVyt32JN0.oit8IbH3zpvaCKQb-N-v7YUxhRmNtzx_TsYYwsKMZn-QdJBHyAsLyUgiiQP081TDze2D3FeATNgEjDsIj2KYWQ&_tp=eyJjb250ZXh0Ijp7ImZpcnN0UGFnZSI6InB1YmxpY2F0aW9uIiwicGFnZSI6InB1YmxpY2F0aW9uIiwicG9zaXRpb24iOiJwYWdlSGVhZGVyIn19) and attempted to use multi-DVGeo to fix this issue. I followed the offered test case to add an extra block (named block2) to wrap the cylindrical. 
![image](https://github.com/user-attachments/assets/8e4fd794-015f-4f6a-bfa7-21be09c3f1b2)
![image](https://github.com/user-attachments/assets/0b74b31b-caa7-4eb2-ad33-4a8b445a151e)

I used addPointSet() to add the mesh to multi-DVGeo but it raise an error of "Error: The point at (x, y, z) = (51.489, -1.686 15.205) in point set coords is inside multiple FFDs but a triangulated mesh for component block1 is not provided to determine which component owns this point''. I didn't see it was added a compName to addPointSet in test case, but I still tried assigning a block name to compNames in addPointSet, but another error raised as "Error: The point at (x, y, z) = (52.093, -2.044 15.502) in point set coords is not inside any FFDs". I checked the mesh and I confirmed that the surface mesh is enclosed within the FFD blocks. How can I resolve this issue? or is there is another example to show how to use DVGeo-multi? Thank you! Below is my test code.

gridFile='bump_cylindrical_volume_test.cgns' #this is a single overset volume mesh
aeroOptions = {"gridFile": gridFile, }
meshOptions = {"gridFile": gridFile,'LdefFact':50.0}
mesh = USMesh(options=meshOptions)
coords = mesh.getSurfaceCoordinates()
comm = MPI.COMM_WORLD
FFDFile = "block1.xyz"
FFDFile2= 'blcok2.xyz'
DVGeo3 = DVGeometry(FFDFile,isComplex=False)
DVGeo4 = DVGeometry(FFDFile2,isComplex=False)
DVGeo = DVGeometryMulti(comm=comm)
DVGeo.addComponent('block1',DVGeo3)
DVGeo.addComponent('block2',DVGeo4)
DVGeoDict = DVGeo.getDVGeoDict()
ps4=geo_utils.PointSelect(psType='x',pt1=[49, 4, 14.4],pt2=[65 ,-4 ,14.6])
DVGeoDict['block1'].addLocalDV("local1x", lower=-1, upper=1, axis="x", pointSelect=ps1,scale=1)
DVGeoDict['block1'].addLocalDV("local1y", lower=-1, upper=1, axis="y", pointSelect=ps1,scale=1)
DVGeo.addPointSet(coords, "coords",compNames=['block1'], applyIC=True)
DVGeo.addIntersection(compA='block1',compB='block2',project=True)
dvDict = DVGeo.getValues()
dvDict["local1x"][:] = 2
DVGeo.setDesignVars(dvDict)
coords_new = DVGeo.update("coords")
DVGeo.writePlot3d("ffd_deformed.xyz")
DVGeo.writePointSet("coords", "surf")
mesh.setSurfaceCoordinates(coords_new)
mesh.warpMesh()
mesh.writeGrid('volume_warped.cgns')

**Comment by hajdik** [comment]
See [the tutorial](https://mdolab-mach-aero.readthedocs-hosted.com/en/latest/machAeroTutorials/intersection_overview.html) for DVGeoMulti for info on setting up the intersection system.
